envs/monkey_zoo/terraform/firewalls.tf

```diff
@@ -96,5 +97,158 @@ resource "google_compute_firewall" "tunneling2-out" {
   }
 
   direction = "EGRESS"
-  destination_ranges = ["10.2.0.0/24"]
+  destination_ranges = ["10.2.0.0/27"]
+}
+
+resource "google_compute_firewall" "allow-tunneling-only-from-islands" {
+  name    = "${local.resource_prefix}allow-tunneling-only-from-islands"
+  network = google_compute_network.monkeyzoo.name
+
+  allow {
+    protocol = "all"
+  }
+
+  direction = "INGRESS"
+  priority = "1000"
+  source_tags = ["island"]
+  target_tags = ["tunneling-9"]
+}
+
+
+resource "google_compute_firewall" "credentials-reuse-in" {
+  name    = "${local.resource_prefix}credentials-reuse-in"
+  network = google_compute_network.credential-reuse.name
+
+  allow {
+    protocol = "all"
+  }
+
+  direction = "INGRESS"
+  priority = "1000"
+  source_ranges = ["0.0.0.0/0"]
+}
+
+resource "google_compute_firewall" "credential-reuse-allow-icmp" {
+  name    = "${local.resource_prefix}credential-reuse-allow-icmp"
+  network = google_compute_network.credential-reuse.name
+
+  allow {
+    protocol = "icmp"
```

Comment:
How is this different from the rule above? I think icmp is included in "all" protocols